inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_proposal_to_proposal_ie.cpp

```diff
@@ -8,58 +8,70 @@
 #include <vector>
 
 #include <ngraph/opsets/opset1.hpp>
-
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph_ops/proposal_ie.hpp>
 #include <ngraph/rt_info.hpp>
 
-ngraph::pass::ConvertProposalToLegacyMatcher::ConvertProposalToLegacyMatcher() {
-    auto input_0 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_1 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_2 = std::make_shared<pattern::op::Label>(element::f32, Shape{3});
-
-    ngraph::op::ProposalAttrs attr = {};
+bool convert_to_proposal_ie(std::shared_ptr<ngraph::op::v0::Proposal> proposal, bool infer_probs = false) {
+    ngraph::Output<ngraph::Node> last; // 2D tensor of size [1, 3-4] with im_info will be retrieved from this node
+    ngraph::NodeVector ops_to_replace, new_ops;
+    ops_to_replace.push_back(proposal);
+
+    if (auto reshape = std::dynamic_pointer_cast<ngraph::opset1::Reshape>(proposal->input_value(2).get_node_shared_ptr())) {
+        const ngraph::PartialShape& im_info_shape = reshape->get_input_partial_shape(0);
+        NGRAPH_CHECK(im_info_shape == ngraph::Shape({1, 3}) || im_info_shape == ngraph::Shape({1, 4}));
```

Comment:
We do not throw exceptions inside transformations. Use return false

Reply:
Done.